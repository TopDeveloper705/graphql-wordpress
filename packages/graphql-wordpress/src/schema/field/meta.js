import { GraphQLList } from 'graphql';
import Meta from 'type/Meta';

const metaResolver = data => {
  if (!data.meta || !Object.keys(data.meta).length) {
    return null;
  }
  return Object.keys(data.meta).map(key => ({
    name: key,
    value: data.meta[key],
  }));
};

export default () => ({
  type: new GraphQLList(Meta),
  description: 'Meta fields.',
  resolve: metaResolver,
});
