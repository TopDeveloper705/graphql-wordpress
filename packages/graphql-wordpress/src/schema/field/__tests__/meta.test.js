import meta from 'field/meta';

describe('Test schema type field definition', () => {
  const field = meta();

  test('Test meta field', () => {
    expect(field.type).toMatchSnapshot();
  });

  test('Test meta field resolver', () => {
    expect(field.resolve({ meta: { taco: 'bell', doritos: 'loco' } })).toMatchSnapshot();
  });

  test('Test meta field resolver null', () => {
    expect(field.resolve({ meta: null })).toBeNull();
  });
});
