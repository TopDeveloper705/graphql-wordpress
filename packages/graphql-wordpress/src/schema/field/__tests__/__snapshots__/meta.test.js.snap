// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test schema type field definition Test meta field 1`] = `"[Meta]"`;

exports[`Test schema type field definition Test meta field resolver 1`] = `
Array [
  Object {
    "name": "taco",
    "value": "bell",
  },
  Object {
    "name": "doritos",
    "value": "loco",
  },
]
`;
